### PR TITLE
fix some warnings and clean up some typing issues

### PR DESCRIPTION
**Pull request by gavinking**
This is a general cleanup of the Session + Query hierarchies, focussing on stuff the IDE was complaining about..

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/SharedSessionContract.java [hunk 1]
```diff
@@ -18,18 +18,15 @@
...
- * <p/>
- * NOTE : Poorly named.  "shared" simply indicates that its a unified contract between {@link Session} and
- * {@link StatelessSession}.
- * 
+ * Declares operations that are common between {@link Session} and {@link StatelessSession}.
+ *
  * @author Steve Ebersole
  */
-public interface SharedSessionContract extends QueryProducer, Serializable {
+//NOTE : This interface is poorly named: "shared" simply indicates that it's a unified contract
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/SharedSessionContract.java:28]: Please remove this line.  We had a very long and drawn out discussion and consensus about the naming here.  Whether you like it or not, you had the chance to chime in when we had that original discussion ;)

File: hibernate-core/src/main/java/org/hibernate/engine/spi/SessionImplementor.java [hunk 2]
```diff
@@ -26,7 +25,8 @@
  * {@link org.hibernate.type.Type}, {@link EntityPersister}
  * and {@link org.hibernate.persister.collection.CollectionPersister} implementations.
  *
- * A Session, through this interface and SharedSessionContractImplementor, implements:<ul>
+ * A Session, through this interface and SharedSessionContractImplementor, implements:
+ * <ul>
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/engine/spi/SessionImplementor.java:29]: Minor so don't worry about it here, but in the future this was the preferred syntax for html lists

File: hibernate-core/src/main/java/org/hibernate/engine/spi/SessionImplementor.java [hunk 3]
```diff
@@ -157,4 +117,44 @@ default SessionImplementor getSession() {
...
+	@Override
+	<T> QueryImplementor<T> createQuery(CriteriaQuery<T> criteriaQuery);
+
+	@Override @SuppressWarnings("rawtypes")
+	QueryImplementor createQuery(CriteriaUpdate updateQuery);
+
+	@Override @SuppressWarnings("rawtypes")
+	QueryImplementor createQuery(CriteriaDelete deleteQuery);
+
+//		@Override @SuppressWarnings("rawtypes")
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/engine/spi/SessionImplementor.java:131]: Please remove this commented out block

File: hibernate-core/src/main/java/org/hibernate/query/Query.java [hunk 4]
```diff
@@ -640,40 +640,6 @@ default R getSingleResult() {
 	 */
 	<P> Query<R> setParameterList(int position, Collection<? extends P> values, AllowableParameterType<P> type);
 
-//	/**
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/query/Query.java:643]: Like you did here ;)

**Review by sebersole — COMMENTED** [review]
Left a few comments, other than that.. lgtm

File: hibernate-core/src/main/java/org/hibernate/SharedSessionContract.java (hunk 1 above)
Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/SharedSessionContract.java:28]: I didn't add that line, it was already there, I just moved it out of the Javadoc and into a regular comment.

Do you want me to put it back into the Javadoc? Or just delete it altogether (my preference).

Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/SharedSessionContract.java:28]: Just remove it

File: hibernate-core/src/main/java/org/hibernate/query/spi/AbstractQuery.java [hunk 5]
```diff
@@ -1403,8 +1403,7 @@ else if ( retType.isArray() ) {
 	}
 
 	@Override
-	@SuppressWarnings( "rawtypes" )
-	public QueryImplementor<R> setProperties(Map map) {
+	public QueryImplementor<R> setProperties(Map<?,?> map) {
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/query/spi/AbstractQuery.java:1406]: I always think inserting a space char between `?,?` improves code readability (and in line with our existing coding format). 
```
public QueryImplementor<R> setProperties(Map<?, ?> map) {
... ...
}
```
I always hesitated whether nailing down the first type generics to String is better or not
```
public QueryImplementor<R> setProperties(Map<String, ?> map) {
```

Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/query/spi/AbstractQuery.java:1406]: While we don't have a specific code guideline around this, I personally prefer the no space version as well until these get "nested".

But I'm certainly not going to ask someone to make that change in a PR

**Pull request merged**